hs_composite_resource/models.py

```diff
@@ -95,11 +95,15 @@ def get_file_aggregation_object(self, file_path):
          :param file_path: Resource file path (full file path starting with resource id)
          for which the aggregation object to be retrieved
         """
-        for res_file in self.files.all():
-            if res_file.full_path == file_path:
-                if res_file.has_logical_file:
-                    return res_file.logical_file
-        return None
+        relative_file_path = file_path[len(self.file_path) + 1:]
```

Comment:
This block is where we get the largest speedup, right?  Instead of looping through each file in the resource, we're doing a db lookup.

Reply:
That's correct.